File: Javascript/InsertionSort.js
const arr = [8, 5, 7, 3, 2];

function Insertion_sort(A) {
  //length of array
  const n = arr.length;
  // for pass
  for (let i = 1; i < n; i++) {
    //for comparison and swaping
    let j = i - 1;
    let el = A[i];
    while (j > -1 && A[j] > el) {
      A[j + 1] = A[j];
      j--;
    }
    A[j + 1] = el;
  }

  //displaying after sort
  for (let i = 0; i < n; i++) console.log(A[i]);
}

Insertion_sort(arr);

// complexity = O(n^2)
// stable = true
// Adaptive

File: C++/insertionSort.cpp
#include<conio.h>
#include<iostream>

using namespace std;


  void insertionSort(int arr[], int n){
  	for(int i=1;i<n;++i){
  		int j = i-1;
  		int x = arr[i];
  		while(j>-1 && arr[j]> x){
  			arr[j+1] = arr[j];
  			j--;
		}
		arr[j+1] = x;
	  }
  }

  void display(int arr[], int n){
   	   for(int i=0;i<n;++i){
    		cout<<arr[i]<<" ";
    	}
	    cout<<endl;
  }

int main(){
	int arr[] = {8,7,5,3,2};
	int n=5;
	
	//before sort
	cout<<"befor? sort"<<endl;
	display(arr,5);
	
	insertionSort(arr, n);
	
	// after sort
		cout<<"after sort"<<endl;
	display(arr, n);
 }
